File: kamacoder106/kamacoder106.cpp
#include<iostream>
#include<vector>

using namespace std;
int main(){
    int n, m;
    cin >> n >> m;
    vector<vector<int>> grid(n, vector<int>(m, 0));
    for (int i = 0; i < n; i++){
        for (int j = 0; j < m; j++){
            cin >> grid[i][j];
        }
    }

    int direction[4][2] = {0, 1, 1, 0, -1, 0, 0, -1};
    int result = 0;

    for (int i = 0; i < n; i++){
        for (int j = 0; j < m; j++){
            if (grid[i][j] == 1){
                for (int k = 0; k < 4; k++){
                    int x = i + direction[k][0];
                    int y = j + direction[k][1];

                    if (x < 0 || x >= grid.size() || y < 0 || y >= grid[0].size() || grid[x][y] == 0){
                        result++;
                    }
                }
            }
        }
    } 
    cout << result << endl;
}

File: kamacoder106/kamacoder106.py
n, m = map(int, input().split())
grid = []
for i in range(n):
    grid.append(list(map(int, input().split())))

direction = [[1, 0], [-1, 0], [0, 1], [0, -1]]
result = 0

for i in range(n):
    for j in range(m):
        if grid[i][j] == 1:
            for x, y in direction:
                next_x = i + x
                next_y = j + y
                if next_x < 0 or next_y < 0 or next_y >= len(grid[0]) or next_x >= len(grid) or grid[next_x][next_y] == 0:
                    result += 1

print(result)